zgnroot/src/gl-base-technique.c

```diff
@@ -118,10 +119,12 @@ zgnr_gl_base_technique_protocol_draw_arrays(struct wl_client *client,
     struct wl_resource *resource, uint32_t mode, int32_t first, uint32_t count)
 {
   UNUSED(client);
-  UNUSED(resource);
-  UNUSED(mode);
-  UNUSED(first);
-  UNUSED(count);
+  struct zgnr_gl_base_technique_impl *self =
+      wl_resource_get_user_data(resource);
+  self->pending.draw_method = ZGNR_GL_BASE_TECHNIQUE_DRAW_ARRAYS;
```

Comment:
```suggestion
  struct zgnr_gl_base_technique_impl *self =
      wl_resource_get_user_data(resource);
  if (self == NULL) return;
      
  self->pending.draw_method = ZGNR_GL_BASE_TECHNIQUE_DRAW_ARRAYS;
```

Regarding base technique, it's wl_resource can be inert.
Inert resource means, the resoruce object itself is alive but there is no bound object. That is, wl_resource_get_user_data(resource) is NULL.

As for base technique, that happens when rendering unit bound to the base technique is destroyed.

If you need more explanation, feel free to ask me at any time!